codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithydafny/DafnyApiCodegen.java

```diff
@@ -1421,6 +1537,83 @@ private TokenTree OperationMemberModifies(
             )
         )
         .lineSeparated();
+    } else if (isUnion && member.isRequired()) {
+      // Required union item
+      // This is very annoying
+      // the way that the decreases clause work
+      // it needs to delimit this list of values
+      // It treads each return as an individual TokenTree
+      // Without the toString, this will be broken up in the wrong way :(
+      return TokenTree.of(
+        TokenTree
+          .of(
+            Token.of("(match %s ".formatted(varName)),
+            Token.of(memberShape
+              .asUnionShape()
+              .get()
+              .members()
+              .stream()
+              .filter(this::OnlyReferenceStructures)
+              .map(s -> Token
+                .of(
+                  "case %s(o) => o.Modifies"
+                    .formatted(s.getMemberName())
+                ))),
+            TokenTree.of(
+              memberShape
+                .asUnionShape()
+                .get()
+                .members()
+                .stream()
+                .allMatch(this::OnlyReferenceStructures)
+                ? ")"
+                : "case _ => {})")
+          )
+          .flatten()
+          .dropEmpty()
+          .lineSeparated()
+          .toString()
+      );
+    } else if (isUnion && !member.isRequired()) {
+      // Optional union item
+      // Required union item
```

Comment:
```suggestion
```

Reply:
This is the optional case only,
you handled the required case above,
right?

Reply:
fixed